### PR TITLE
ci: update juju to 3.1 and microk8s to 1.25

### PR DESCRIPTION
**Pull request by NohaIhab**
ci: update juju to 3.1 and microk8s to 1.25

### CODE REVIEW

File: charms/argo-controller/requirements-integration.in [hunk 1]
```diff
@@ -1,7 +1,7 @@
 aiohttp
 jinja2
-# Pinning to <3.0 to ensure compatibility with the 2.9 controller version
-juju<3.0
+# Pinning to <4.0 due to compatibility with the 3.1 controller version
+juju<4.0
```
[the remark this answers is not in the document]
Comment (NohaIhab) [reply charms/argo-controller/requirements-integration.in:4]: based on discussions and the post by the Juju team on controller-pyjuju compaiblities, we're pinning to <4.0

File: charms/argo-controller/requirements-integration.txt [hunk 2]
```diff
@@ -64,7 +64,7 @@ jinja2==3.1.2
     # via
     #   -r requirements-integration.in
     #   pytest-operator
-juju==2.9.44.0
+juju==3.2.0.1
```
Comment (phoevos) [inline charms/argo-controller/requirements-integration.txt:67]: This is 3.2.2 everywhere else. Is something else constraining us or are the requirements just older?

**Review by phoevos — CHANGES_REQUESTED** [review]
Thanks @NohaIhab, just an observation re the juju dependency!

File: charms/argo-controller/requirements-integration.txt (hunk 2 above)
Comment (phoevos) [reply charms/argo-controller/requirements-integration.txt:67]: Manually upgrading gives me some other upgrades too:
```diff
diff --git a/charms/argo-controller/requirements-integration.txt b/charms/argo-controller/requirements-integration.txt
index 192ecc8..e6bfc78 100644
--- a/charms/argo-controller/requirements-integration.txt
+++ b/charms/argo-controller/requirements-integration.txt
@@ -48,6 +48,8 @@ frozenlist==1.4.0
     #   aiosignal
 google-auth==2.22.0
     # via kubernetes
+hvac==1.2.0
+    # via juju
 idna==3.4
     # via
     #   requests
@@ -64,18 +66,14 @@ jinja2==3.1.2
     # via
     #   -r requirements-integration.in
     #   pytest-operator
-juju==3.2.0.1
+juju==3.2.2
     # via
     #   -r requirements-integration.in
     #   pytest-operator
-jujubundlelib==0.5.7
-    # via theblues
 kubernetes==27.2.0
     # via juju
 macaroonbakery==1.3.1
-    # via
-    #   juju
-    #   theblues
+    # via juju
 markupsafe==2.1.3
     # via jinja2
 matplotlib-inline==0.1.6
@@ -121,6 +119,8 @@ pycparser==2.21
     # via cffi
 pygments==2.16.1
     # via ipython
+pyhcl==0.4.5
+    # via hvac
 pymacaroons==0.13.0
     # via macaroonbakery
 pynacl==1.5.0
@@ -132,7 +132,7 @@ pyrfc3339==1.1
     # via
     #   juju
     #   macaroonbakery
-pytest==7.4.1
+pytest==7.4.2
     # via
     #   pytest-asyncio
     #   pytest-operator
@@ -147,16 +147,15 @@ pytz==2023.3.post1
 pyyaml==6.0.1
     # via
     #   juju
-    #   jujubundlelib
     #   kubernetes
     #   pytest-operator
 requests==2.31.0
     # via
     #   -r requirements-integration.in
+    #   hvac
     #   kubernetes
     #   macaroonbakery
     #   requests-oauthlib
-    #   theblues
 requests-oauthlib==1.3.1
     # via kubernetes
 rsa==4.9
@@ -174,8 +173,6 @@ stack-data==0.6.2
     # via ipython
 tenacity==8.2.3
     # via -r requirements-integration.in
-theblues==0.5.2
-    # via juju
 tomli==2.0.1
     # via
     #   ipdb
@@ -201,7 +198,7 @@ wcwidth==0.2.6
     # via prompt-toolkit
 websocket-client==1.6.2
     # via kubernetes
-websockets==7.0
+websockets==8.1
     # via juju
 yarl==1.9.2
     # via aiohttp
```

Comment (NohaIhab) [reply charms/argo-controller/requirements-integration.txt:67]: will re-run pip-compile there

**Review by phoevos — APPROVED** [review]
As long as the CI is green, I'm good!
For the unresolved discussion take it up with @beliaev-maksim 😇

**Pull request merged**